core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/RemoveRedundantPredicateAboveTableScan.java

```diff
@@ -61,9 +64,7 @@ public class RemoveRedundantPredicateAboveTableScan
 
     private static final Pattern<FilterNode> PATTERN =
             filter().with(source().matching(
-                    tableScan().capturedAs(TABLE_SCAN)
-                            // avoid extra computations if table scan doesn't have any enforced predicate
-                            .matching(node -> !node.getEnforcedConstraint().isAll())));
+                    tableScan().capturedAs(TABLE_SCAN)));
```

Comment:
I wonder if it would better to make a separate Rule which would match for `!filterNode.getPredicate().isAll()` and apply only the new optimization.
This Rule seems to be more about eliminating redundancy between the filter already pushed into the connector and the predicate evaluated by the engine.
@martint wdyt ?